src/commands/inputs.rs

```diff
@@ -0,0 +1,167 @@
+#[command(about = "Generate input JSON from a WDL document", version, about)]
+pub struct InputsArgs {
+    #[arg(required = true)]
+    #[clap(value_name = "input path")]
+    pub document: String,
+
+    #[arg(short, long)]
+    #[clap(value_name = "workflow name")]
+    pub name: Option<String>,
+
+    #[arg(short, long)]
+    #[clap(value_name = "output path")]
+    pub output: Option<PathBuf>,
+
+    #[arg(short, long)]
+    #[clap(value_name = "literal defaults")]
+    pub literal_defaults: bool,
+
+    #[arg(short, long)]
+    #[clap(value_name = "override expressions")]
+    pub override_expressions: bool,
+
+}
+
+pub async fn generate_inputs(args: InputsArgs) -> Result<()> {
+    // println!("{:?}", args);
+
+    let results: Vec<wdl::analysis::AnalysisResult> =
+        analyze(args.document.as_str(), vec![], false, false).await?;
+
+    // println!("{:?}", results);
+
+    let uri: Url = Url::parse(args.document.as_str()).unwrap_or_else(|_| {
+        path_to_uri(args.document.as_str()).expect("file should be a local path")
+    });
+
+    // println!("{:?}", uri);
+
+    let result = results
+        .iter()
+        .find(|r| **r.document().uri() == uri)
+        .context("failed to find document in analysis results")?;
+
+    let document = result.document();
+
+    let diagnostics = document.diagnostics();
+    if !diagnostics.is_empty() {
+        for diagnostic in diagnostics {
+            anyhow::bail!("Failed to parse WDL document: {:?}", diagnostic);
+        }
+    }
+
+    let (_path, name, inputs) = wdl::cli::parse_inputs(document, Some("main"), None)?;
+
+    println!("name: {:?}    {:?}", name, inputs);
+
+    // search the document to match a task or workflow by name
+    let input_section: &IndexMap<String, wdl::analysis::document::Input> = if let Some(name) =
+    args.name
+    {
+        if let Some(task) = document.task_by_name(&name) {
+            task.inputs()
+        } else if let Some(workflow) = document.workflow() {
+            if workflow.name() == name {
+                workflow.inputs()
+            } else {
+                anyhow::bail!("No task or workflow found with name '{}'", name);
+            }
+        } else {
+            anyhow::bail!("No task or workflow found with name '{}'", name);
+        }
+    } else {
+        // If no name is provided, try workflow first
+        if let Some(workflow) = document.workflow() {
+            workflow.inputs()
+        } else {
+            // No workflow - look for exactly one task
+            let tasks: Vec<_> = document.tasks().collect();
+            println!("tasks: {:?}", tasks);
+            match tasks.len() {
+                0 => anyhow::bail!("No workflow or tasks found in document"),
+                1 => tasks[0].inputs(),
+                _ => anyhow::bail!(
+                    "Multiple tasks found in document but no name specified. Please provide a name using --name"
+                ),
+            }
+        }
+    };
```

Comment:
this should all be handled internally by `cli::parse_inputs()`, so you shouldn't have to do any of this

Reply:
This actually looks correct! I should have been paying closer attention earlier. Ignore my advice to use `parse_inputs()`, you were doing it right!